third_party/move/move-compiler-v2/src/file_format_generator/peephole_optimizer/inefficient_binops.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module contains a fixed window peephole optimizer for the Move bytecode.
+//! As with all peephole optimizers here, it assumes that the bytecode is valid.
+//!
+//! We consider fixed windows of size 5 for this optimizer.
+//!
+//! This optimizer addresses a commonly appearing pattern when compiling binary
+//! operations where the second operand to the operation is a constant.
+//!
+//! The pattern is:
+//! 1. Load the constant (second operand) on the stack.
```

Comment:
Can't this be easily generalized to a simpler pattern:
```
    load constant;
    store to local i;
    ... <any code not using i>
    move i to stack;
```
optimizes to
```
    ... <any code not using i>
    load constant
```

Reply:
Good idea! This is not seen much in the aptos-framework code (among 30k instructions), but I was able to create a simple test case where we do see it (I have added the test case to this PR).

I have added a feature request to implement it as a follow up: https://github.com/aptos-labs/aptos-core/issues/14169 (see text for more information).